services/property-svc/stories/GetPropertieBySubjectType_test.go

```diff
@@ -0,0 +1,163 @@
+
+	//
+	// Create new Properties
+	//
+
+	createPropertyRequestA := &pb.CreatePropertyRequest{
+		SubjectType:   pb.SubjectType_SUBJECT_TYPE_TASK,
+		FieldType:     pb.FieldType_FIELD_TYPE_TEXT,
+		Name:          t.Name() + " A",
+		Description:   nil,
+		SetId:         nil,
+		FieldTypeData: nil,
+	}
+
+	createResponseA, err := propertyClient.CreateProperty(ctx, createPropertyRequestA)
+	if !assert.NoError(t, err, "could not create new property A") {
+		return
+	}
+	_, err = uuid.Parse(createResponseA.PropertyId)
+	if !assert.NoError(t, err, "propertyIDA is not a uuid") {
+		return
+	}
+
+	createPropertyRequestB := &pb.CreatePropertyRequest{
+		SubjectType: pb.SubjectType_SUBJECT_TYPE_PATIENT,
+		FieldType:   pb.FieldType_FIELD_TYPE_SELECT,
+		Name:        t.Name() + " B",
+		Description: nil,
+		SetId:       nil,
+		FieldTypeData: &pb.CreatePropertyRequest_SelectData_{
+			SelectData: &pb.CreatePropertyRequest_SelectData{
+				AllowFreetext: hwutil.PtrTo(false),
+				Options: []*pb.CreatePropertyRequest_SelectData_SelectOption{
+					{
+						Name:        "Option 1",
+						Description: nil,
+					},
+					{
+						Name:        "Option 2",
+						Description: hwutil.PtrTo("Some Description"),
+					},
+				},
+			},
+		},
+	}
+
+	createResponseB, err := propertyClient.CreateProperty(ctx, createPropertyRequestB)
+	if !assert.NoError(t, err, "could not create new property B") {
+		return
+	}
+	_, err = uuid.Parse(createResponseB.PropertyId)
+	if !assert.NoError(t, err, "propertyIDB is not a uuid") {
+		return
+	}
+
+	createPropertyRequestC := &pb.CreatePropertyRequest{
+		SubjectType:   pb.SubjectType_SUBJECT_TYPE_PATIENT,
+		FieldType:     pb.FieldType_FIELD_TYPE_NUMBER,
+		Name:          t.Name() + " C",
+		Description:   nil,
+		SetId:         nil,
+		FieldTypeData: nil,
+	}
+
+	createResponseC, err := propertyClient.CreateProperty(ctx, createPropertyRequestC)
+	if !assert.NoError(t, err, "could not create new property C") {
+		return
+	}
+	_, err = uuid.Parse(createResponseC.PropertyId)
+	if !assert.NoError(t, err, "propertyIDC is not a uuid") {
+		return
+	}
+
+	hwtesting.WaitForProjectionsToSettle()
+
+	//
+	// Get Properties
+	//
+
+	propertiesResponse, err := propertyClient.GetPropertiesBySubjectType(ctx, &pb.GetPropertiesBySubjectTypeRequest{
+		SubjectType: pb.SubjectType_SUBJECT_TYPE_PATIENT,
+	})
+	if !assert.NoError(t, err, "could not get properties after they were created") {
+		return
+	}
+
+	assert.Len(t, propertiesResponse.Properties, 2)
+
+	propertyMap := make(map[string]map[string]interface{})
+
+	for _, property := range propertiesResponse.Properties {
+		s, err := json.Marshal(property)
+		assert.NoError(t, err)
+
+		re := regexp.MustCompile(`[a-f0-9]{8}-[a-f0-9]{4}-[1-5][a-f0-9]{3}-[89ab][a-f0-9]{3}-[a-f0-9]{12}`)
+		res := re.ReplaceAllString(string(s), "<id>")
+
+		v := make(map[string]interface{})
+		err = json.Unmarshal([]byte(res), &v)
+		assert.NoError(t, err)
+
+		propertyMap[property.Id] = v
```

Comment:
Why removing the IDs?

Reply:
because we dont have the subtask ids to compare to